File: src/lessons/welcome/methods/slug/SlugHuntingEntity.java
package lessons.welcome.methods.slug;

import java.awt.Color;


public class SlugHuntingEntity extends plm.universe.bugglequest.SimpleBuggle {

	@Override
	public void run() {
		hunt(); 
	}

	/* BEGIN TEMPLATE */
	public void hunt() {
		// Write your code here
		/* BEGIN SOLUTION */
		while (! isOverBaggle()) {
			if (isFacingTrail()) {
				brushDown();
				stepForward();
				brushUp();
			} else {
				left();
			}
		}
		pickupBaggle();
	}

	boolean isFacingTrail() {
		if (isFacingWall())
			return false;

		stepForward();
		boolean res = getGroundColor()== Color.green;
		stepBackward();
		return res;

		/* END SOLUTION */
	}		
	// Copy your isFacingTrail here
	/* END TEMPLATE */


}

File: src/lessons/welcome/methods/slug/SlugHuntingEntity.py
import java.awt.Color as Color

# BEGIN TEMPLATE
# Copy your isFacingTrail, and write your code below
# BEGIN SOLUTION
def isFacingTrail():
   if isFacingWall():
      return False
   else:
	  forward()
	  res = getGroundColor() == Color.green
	  backward()
	  return res

while not isOverBaggle():
   brushUp()
   if isFacingTrail():
      brushDown()
      forward()
      brushUp()
   else:
     left()
pickupBaggle()
# END SOLUTION
# END TEMPLATE
